src/app/util/ember-compatibility-functions.cpp

```diff
@@ -575,36 +556,66 @@ CHIP_ERROR WriteSingleClusterData(ClusterInfo & aClusterInfo, TLV::TLVReader & a
     attributePathParams.mClusterId   = aClusterInfo.mClusterId;
     attributePathParams.mAttributeId = aClusterInfo.mAttributeId;
 
+    Protocols::InteractionModel::Status status = Protocols::InteractionModel::Status::Failure;
+
     // Named aPath for now to reduce the amount of code change that needs to
     // happen when the above TODO is resolved.
     ConcreteDataAttributePath aPath(aClusterInfo.mEndpointId, aClusterInfo.mClusterId, aClusterInfo.mAttributeId);
-    const EmberAfAttributeMetadata * attributeMetadata =
-        emberAfLocateAttributeMetadata(aPath.mEndpointId, aPath.mClusterId, aPath.mAttributeId, CLUSTER_MASK_SERVER, 0);
-
-    if (attributeMetadata == nullptr)
     {
-        return apWriteHandler->AddStatus(attributePathParams, Protocols::InteractionModel::Status::UnsupportedAttribute);
-    }
+        AttributeAccessInterface * attrOverride = findAttributeAccessOverride(aClusterInfo.mEndpointId, aClusterInfo.mClusterId);
+        if (attrOverride != nullptr)
+        {
+            AttributeValueDecoder valueDecoder(aReader, apWriteHandler->GetAccessingFabricIndex());
+            ReturnErrorOnFailure(attrOverride->Write(aPath, valueDecoder));
 
-    if (attributeMetadata->IsReadOnly())
-    {
-        return apWriteHandler->AddStatus(attributePathParams, Protocols::InteractionModel::Status::UnsupportedWrite);
+            if (valueDecoder.TriedDecode())
+            {
+                status = Protocols::InteractionModel::Status::Success;
+                ExitNow();
+            }
+        }
     }
 
-    AttributeAccessInterface * attrOverride = findAttributeAccessOverride(aClusterInfo.mEndpointId, aClusterInfo.mClusterId);
-    if (attrOverride != nullptr)
     {
-        AttributeValueDecoder valueDecoder(aReader, apWriteHandler->GetAccessingFabricIndex());
-        ReturnErrorOnFailure(attrOverride->Write(aPath, valueDecoder));
+        const EmberAfAttributeMetadata * metadata =
+            emberAfLocateAttributeMetadata(aPath.mEndpointId, aPath.mClusterId, aPath.mAttributeId, CLUSTER_MASK_SERVER, 0);
```

Comment:
This, and the null-check for its result, needs to move higher.

Reply:
It's at the top of the function now.